### PR TITLE
[iOS] Implement the calendar registration feature on the session detail

### LINKED ISSUE #887: [iOS] Add calendar registration feature on timetable detail screen
## Summary
Now we have no functionality about calendar registration sessions. Let's make it possible to register session information to a calendar.

### PR DESCRIPTION
**Pull request by woxtu**
## Issue
- close #887

## Overview (Required)
- This PR implements the calendar registration feature on the session detail
  - The event module is copied from the official app 2022
  - This PR doesn't provide any localizations

## Links
- https://github.com/DroidKaigi/conference-app-2022/tree/main/app-ios/Sources/Event

## Screenshot (Optional if screenshot test is present or unrelated to UI)
Before | After
:--: | :--:
 | 

## Movie (Optional)
Before | After
:--: | :--:
<video src="" width="300" > | <video src="" width="300" >

### CODE REVIEW

File: app-ios/Modules/Sources/Session/SessionViewModel.swift [hunk 1]
```diff
@@ -1,10 +1,37 @@
+import Dependencies
+import Event
 import Foundation
 import shared
 
-final class SessionViewModel {
+@MainActor
+final class SessionViewModel: ObservableObject {
+    @Dependency(\.eventKitClient) var eventKitClient
+    @Published var isAddingToCalendarConfirming: Bool = false
```
Comment (ry-itto) [inline app-ios/Modules/Sources/Session/SessionViewModel.swift:9]: This is nits! I think this flag represents dialog presenting state, so `@State` might be better 👀 
How about this?

**Review by ry-itto — APPROVED** [review]
Almost LGTM!
I left a nits comment. Please check it 🙏🏼

File: app-ios/Modules/Sources/Session/SessionViewModel.swift (hunk 1 above)
Comment (woxtu) [reply app-ios/Modules/Sources/Session/SessionViewModel.swift:9]: My understanding is that a `@State` on an `ObservableObject` doesn't update any views.

Comment (ry-itto) [reply app-ios/Modules/Sources/Session/SessionViewModel.swift:9]: @woxtu 
I didn't have enough to say. My suggestion is to use `@State` in `View` and use that 🙏🏼

Comment (woxtu) [reply app-ios/Modules/Sources/Session/SessionViewModel.swift:9]: The view model takes its binding?

Comment (woxtu) [reply app-ios/Modules/Sources/Session/SessionViewModel.swift:9]: Okay.

**Pull request merged**